e2e_test/error_ui/simple/recovery.slt

```diff
@@ -12,19 +12,25 @@ statement ok
 insert into t values (1);
 
 # Wait for recovery to complete.
-sleep 10s
+sleep 15s
 
 # Check that there's a log entry for barrier collection failure on the mata node.
-# The message should contain the root cause of the failure: "Division by zero".
+# Extract the message to check that it contains the root cause of the failure.
 query T
-select info::varchar like
-'%Actor % exited unexpectedly: Executor error: Chunk operation error: Division by zero%'
-from rw_catalog.rw_event_logs
-where event_type = 'COLLECT_BARRIER_FAIL'
-order by timestamp desc
-limit 1;
+with error as (
+    select info->'collectBarrierFail'->'error' #>> '{}' as error
+    from rw_catalog.rw_event_logs
+    where event_type = 'COLLECT_BARRIER_FAIL'
+    order by timestamp desc
+    limit 1
+)
+select (regexp_matches(
+    error,
+    'Actor \d+ exited unexpectedly: (.*);'
+))[1] as error
+from error;
 ----
-t
+Executor error: Chunk operation error: Division by zero
```

Comment:
Maybe we can do sometihng like `case when` for the regexp matches.

If no match, print original error from the error table.

Reply:
What a genius you're! 😁